src/ring_theory/fractional_ideal.lean

```diff
@@ -477,6 +477,13 @@ instance comm_semiring : comm_semiring (fractional_ideal S P) :=
   ..fractional_ideal.has_one,
   ..fractional_ideal.has_mul }
 
+lemma coe_ideal_pow (I : ideal R) (n : ℕ) : (↑(I^n) : fractional_ideal S P) = I^n :=
```

Comment:
It would be nicer if this were true by definition, which you can achieve by adding `npow` to the fields in `ring` above.

Reply:
Do you mean something like defining
```
lemma fractional_pow (n : ℕ) (I : fractional_ideal S P) : is_fractional S (I^n : submodule R P) :=
sorry
@[irreducible]
def pow (n : ℕ) (I : fractional_ideal S P) : fractional_ideal S P := ⟨I^n, fractional_pow n I⟩
```

and setting npow equal to this in the `comm_semiring` instance? How would that make the lemma true by definition?

Reply:
Yes exactly, although I don't think you need irreducibility for `pow` (and that stops it being true "by definition"):
```lean
lemma fractional_pow (n : ℕ) (I : fractional_ideal S P) : is_fractional S (I^n : submodule R P) :=
sorry

instance : has_pow (fractional_ideal S P) ℕ := ⟨λ I n, ⟨I^n, fractional_pow n I⟩
```
then set `npow := (^)` in the `comm_semiring` definition.

Reply:
Ah sorry, I read the coercion wrong, this is the coercion from ideals, not to submodules.

Reply:
Sorry for the confusion, the change I was suggesting is orthogonal, I've made it in #12080